### DISCUSSION TITLE
RTranslator 2.0.0

### DISCUSSION THREAD

**Discussion by niedev** [niedev/RTranslator#6] [General]
The Google API's have been replaced by [Meta's NLLB](https://ai.meta.com/research/no-language-left-behind/) for translation and [OpenAi's Whisper](https://openai.com/index/whisper/) for speech recognition. These AI models run directly on your phone, so now the app is totally free and with no configuration required!

A classic text translation mode has been added.

Bluetooth LE device search has been improved.

Fixed some bugs.

<hr /><em>This discussion was created from the release <a href='https://github.com/niedev/RTranslator/releases/tag/2.0.0'>RTranslator 2.0.0</a>.</em>

**Comment by PawelGu** [comment]
It's... amazing. Really the first app on Android delivering Whisper for offline translation, it works very well (although, naturally, for quick responses rather good hardware is needed) - thank you!!!

**Reply by niedev** [reply]
Thank you! I'm happy that you like it. For now I've tried to optimize the speed as much as possible, but I'm sure that in the future there will be new techniques that I can implement to optimize the execution speed even more (e.g. 4bit quantization on OnnxRuntime, support for GPU inference, the release of [Whisper Distilled](https://github.com/huggingface/distil-whisper) for all languages, or new faster models).

**Comment by kareldonk** [comment]
This is amazing. Incredibly useful. Works fast enough on midrange Samsung A54.  Keep up the great work man.

**Reply by niedev** [reply]
Thank you very much!

**Reply by kareldonk** [reply]
For some reason I could not donate with PayPal. Says it has been denied. Do you have another way, perhaps send money via email with Paypal?

**Reply by kareldonk** [reply]
I used the link on the Github readme page btw.

**Reply by niedev** [reply]
You would be my first donator! Maybe there is something broken with the link, I will fix it ASAP.

**Reply by kareldonk** [reply]
Ok let me know. Honored to be the first :)

**Reply by niedev** [reply]
I tested the link and it should work, however, this is the Paypal link: https://www.paypal.me/lucamartino181
Thank you! :)

**Reply by kareldonk** [reply]
It was an issue at my end. Sorted out and should have gone trhough now. Sorry about that. :)

**Reply by niedev** [reply]
Don't worry, received, thank you so much for the support! 🎉🎉

**Comment by fishergz** [comment]
My Mi 10 is responding very slowly, could it be that I have to get a new phone?

8GB RAM
CPU: Qualcomm Snapdragon 865
Number of CPU cores: Eight
CPU frequency: 1x2.84 GHz A77 + 3x2.42 GHz A77 + 4x1.8 GHz A55
GPU: Adreno 650

**Reply by niedev** [reply]
It's a pretty old phone, so unfortunately it's normal that it's quite slow on RTranslator (to have a good enough quality I had to choose some quite heavy models). 

The good thing is that if you were looking for an excuse to change your phone this is your chance 😆

**Reply by fishergz** [reply]
Hehe, thank you for your good news.

I don't play games, and I change my phone very slowly. The motivation before was only from the improvement of the camera, but now Rtranslator is making an effort again. Your work is really cool!

Yesterday I went to the mall and saw the Samsung S24. Its offline translation seems to be based on Google Gemini nano, and the call translation is very attractive.

But I don't like the appearance of S24 Ultra (too ugly), Then the camera of S24/S24+ is not good enough.

I wonder how Rtranslator performs on Vivo X100 Ultra/Oppo X7 Ultra/Xiaomi 14 Ultra.

It's a dilemma...

When I got home, I realized that the system translation APP on my old iPhone X also supports offline, but the response is slow.

Last time I used the iOS version of Felo in Kazakhstan, an APP based on ChatGPT, it was fantastic when the network was good, but terrible when the network was bad.

I'm going to Turkey & Caucasus in a few months, looking forward to the improvement of Rtranslator :)

The Great Babel Tower Reconstruction Project, you have already started.

**Reply by niedev** [reply]
Thank you very much! I'm sure it will improve in these months 😁 .
Also the smartphones you mentioned are very powerful, so if you planned to buy those (Jokes aside, I hope not mainly for RTranslator) you should notice much better performance.

**Comment by lpacev1ch** [comment]
Very well!!! Nice job!

**Comment by STEVEN5015** [comment]
For the offline phone users, if I download all the .onnx files, where installation path to place these files?Thank you!
